File: spikes/language/ts-bun/src/startup.ts
// Minimal CLI that just prints a message and exits.
// Used to measure cold-start overhead of a Bun/TS process.
const args = process.argv.slice(2);

if (args[0] === "--version") {
  console.log("spike-startup 0.1.0");
} else {
  console.log(
    JSON.stringify(
      {
        tool: "ts-bun",
        operation: "startup",
        status: "ok",
      },
      null,
      2
    )
  );
}

File: spikes/language/rust/src/startup.rs
use std::env;

fn main() {
    // Minimal CLI that just prints a message and exits.
    // Used to measure cold-start overhead of a Rust binary.
    let args: Vec<String> = env::args().collect();
    if args.len() > 1 && args[1] == "--version" {
        println!("spike-startup 0.1.0");
    } else {
        println!("{{");
        println!("  \"tool\": \"rust\",");
        println!("  \"operation\": \"startup\",");
        println!("  \"status\": \"ok\"");
        println!("}}");
    }
}
